Update css block regex to allow final semicolon omission

VivaLaVia opened the pull request:
Update the css regex to accept css blocks that do not have a final
semicolon on the last rule. In cases where a minified stylesheet is
used, its common that this semicolon is omitted.

Associated issues: #3031

**Checklist**

Author:

- [ ] The proper base branch has been selected.  New features go on `unstable`.  Bug-fix patches can go on either `unstable` or `master`.
- [ ] Automated tests have been included in this pull request, if possible, for the new feature(s) or bug fix.  Check this box if tests are not pragmatically possible (e.g. rendering features could include screenshots or videos instead of automated tests).
- [ ] The associated GitHub issues are included (above).
- [ ] Notes have been included (above).

Reviewers:

- [ ] All automated checks are passing (green check next to latest commit).
- [ ] At least one reviewer has signed off on the pull request.
- [ ] Just after this pull request is merged, it should be applied to both the `master` branch and the `unstable` branch.  Normally, this just requires cherry-picking the corresponding merge commit from `master` to `unstable` -- or vice versa.

maxkfranz commented:
It looks like the documentation should also be updated to include your new behaviour.  It currently says this:

> String format
> 
> Note that the trailing semicolons for each property are mandatory. Parsing will certainly fail without them.

https://js.cytoscape.org/#style/format

VivaLaVia commented:
Updated the documentation and added some test for happy/unhappy paths for string sheet semicolon parsing. Please let me know if I goofed any merge conflictions with the changes from master. If needed, I can recreate and fork from the correct branch

maxkfranz commented:
> Updated the documentation and added some test for happy/unhappy paths for string sheet semicolon parsing. 

Great

> Please let me know if I goofed any merge conflictions with the changes from master. If needed, I can recreate and fork from the correct branch

That would be helpful.  Maybe rebasing your branch to the tip of unstable would clean it up without having to start all over and cherry-pick everything

VivaLaVia commented:
Should be all set now. Let me know if you need any further changes

maxkfranz commented:
Nice.

This is now in the unstable branch, which will be released with [3.22.0 on 5 July 2022](https://github.com/cytoscape/cytoscape.js/milestone/226).
